db/batchlog_manager.cc

```diff
@@ -251,7 +251,8 @@ future<> db::batchlog_manager::replay_all_failed_batches() {
                 batch_result.get();
             } catch (data_dictionary::no_such_keyspace& ex) {
                 // should probably ignore and drop the batch
-            } catch (...) {
+            } catch (std::exception& ex) {
```

Comment:
Why narrow the type of exceptions? I know that in theory all exception types we use should inherit from `std::exception` anyway, but still, this leaves a theoretical gap in exception types that can escape the try-catch block.

Reply:
A better and more generic way to log exceptions is to:
```c++
catch (...) {
    mylogger.error("my operation failed because {}", std::current_exception());
}
```
We have a pretty printer for `std::exception_ptr`, which is better than `e.what()` because it also includes the type name.

Reply:
TIL :)